java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/debugging/launch/NbLaunchDelegate.java

```diff
@@ -632,11 +656,27 @@ static Lookup createTargetLookup(Project prj, SingleMethod singleMethod, FileObj
             Lookup methodLookup = Lookups.singleton(singleMethod);
             arr.add(methodLookup);
         }
+        if (nestedClass != null) {
+            Lookup nestedClassLookup = Lookups.singleton(nestedClass);
+            arr.add(nestedClassLookup);
+        }
+        if (projectFilter != null) {
+            Lookup projectLookup = Lookups.singleton(projectFilter);
+            arr.add(projectLookup);
+        }
         if (toRun != null) {
             arr.add(toRun.getLookup());
         }
         return new ProxyLookup(arr.toArray(new Lookup[0]));
     }
+    
+    static ContainedProjectFilter getProjectFilter(Project prj, Map<String, Object> launchArguments) {
+        List<String> projectsArg = argsToStringList(launchArguments.get("projects"));
+        List<Project> projects = ProjectUtils.getContainedProjects(prj, false).stream()
```

Comment:
Note: this does not allow to specify projects nested more levels in the project's hierarchy. Not a bug, and can be enhanced later in a compatible way.